specs/AppLifecycle/Restart/restartApi.markdown

```diff
@@ -0,0 +1,259 @@
+</tr>
+<tr class="odd">
+<td>public static IAsyncOperation&lt;AppRestartFailureReason&gt; <strong>RequestRestartAsync</strong>(<br />
+string launchArguments);</td>
+<td><p>Static function in CoreApplication. Requests immediate termination and restart.</p>
+<ul>
+<li>
+<p>The launchArguments string is passed into the new instance as the Arguments property on LaunchActivatedEventArgs.</p>
+</li>
+<li>
+<p>Also sets LaunchActivatedEventArgs.PreviousExecutionState to Terminated.</p>
+</li>
+</ul></td>
+</tr>
+<tr class="even">
+<td>public enum <strong>AppRestartFailureReason</strong></td>
+<td><ul>
+<li>
+<p>0: RestartPending - a restart is already in progress.</p>
+</li>
+<li>
+<p>1: NotInForeground - an app must be visible and in the foreground when it calls the restart API.</p>
+</li>
+<li>
+<p>2: InvalidUser - could not restart for the specified user.</p>
+</li>
+<li>
+<p>3: Other - unspecified failure.</p>
+</li>
+</ul></td>
+</tr>
+<tr class="odd">
+<td>class <strong>LaunchActivatedEventArg</strong></td>
+<td><p>Relevant Properties:</p>
+<ul>
+<li><p>Arguments: gets the arguments that are passed to the app during its launch activation.</p></li>
+<li><p>Kind: Gets the reason that this app is being activated. This is of type enum ActivationKind</p></li>
+</ul>
+<p>PreviousExecutionState: Gets the execution state of the app before this activation.</p></td>
+</tr>
+</tbody>
+</table>
+
+## New API Details
+
+### RequestRestartNow
+
+<table>
+<colgroup>
+<col style="width: 41%" />
+<col style="width: 53%" />
+<col style="width: 5%" />
+</colgroup>
+<thead>
+<tr class="header">
+<th>New API </th>
+<th>Description</th>
+</tr>
+</thead>
+<tbody>
+<tr class="odd">
+<td>static AppRestartFailureReason <strong>RequestRestartNow</strong>(string restartArguments)</td>
+<td><p>Equivalent to CoreApplication.RequestRestartAsync, except that it is <em>not</em> async.
+<ul>
+<li>If the restart fails, but the user subsequently launches the app manually, the app will launch normally and no restart arguments will be passed. </li>
+<li>If the app has any in-process background tasks running when it calls this API, those tasks will be cancelled in the normal way. Out-of-process background tasks will not be affected. </li>
+</ul>
+ The restarted instance will be activated with Windows.ApplicationModel.Activation.LaunchActivatedEventArgs.</p>
+ <ul>
+<li>When the app is restarted, LaunchActivatedEventArgs.PreviousExecutionState will have the value Terminated so that the app can distinguish between a resume and a restart. </li>
+</ul>
+<p>If the request to restart is successful, the application will be restarted immediately. The user does not have to do anything else/handle a return value in any way.
+
+If the request fails due to a nonfatal reason, a failure reason is returned for why the restart didn’t successfully complete. This method leverages an existing enum (AppRestartFailureReason) which has the following relevant return fields.
```

Comment:
I think this is what I'm asking the samples to show how to use properly. How does a developer correctly use this information to help the app or the end user when something goes wrong?

Reply:
DISCUSSION: There's a nonzero chance that the system couldn't process launch of the agent, or couldn't open the source process to terminate. As long as the API returns one of these codes, doom is imminent.